website/content/docs/nia/configuration.mdx

```diff
@@ -221,6 +221,37 @@ task {
 - `source_includes_var` - (bool: false) Whether or not the module configured at [`task.source`](#source) includes the [`catalog_services` variable](/docs/nia/terraform-modules#catalog-services-variable). Please refer to the documentation of the selected module for guidance on how to configure this field. If configured inconsistently with the module, Consul-Terraform-Sync will error and exit.
 
+
+#### Consul KV Condition
+A consul-kv condition block configures a task to only execute on changes to a Consul KV entry. The condition can be configured for a single Consul KV entry or for any Consul KV entries that are prefixed with a given path.
+
+See [Task Execution: Consul KV Condition](/docs/nia/tasks#consul-kv-condition) for more information on how tasks are triggered with a consul-kv condition.
+
+```hcl
+task {
+  name = "consul_kv_condition_task"
+  description = "execute on changes to Consul KV entry"
+  source = "path/to/consul-kv-module"
+  providers = ["my-provider"]
+  services = ["web-api"]
+
+  condition "consul-kv" {
+    path                = "my-key"
+    recurse             = false
+    datacenter          = "dc1"
+    namespace           = "default"
+    source_includes_var = false
+  }
+}
+```
+
+- `path` - (string: required) The path of the key that is used by the task.
+- `recurse` - (bool: false) Whether or not the path should be treated as a prefix instead of a literal match.
+- `datacenter` - (string) The datacenter of the services to query for the task. If not provided, the datacenter will default to the datacenter of the agent that Consul-Terraform-Sync queries.
+- `namespace` <EnterpriseAlert inline /> - (string) The namespace of the services to query for the task. If not provided, the namespace will be inferred from the Consul-Terraform-Sync ACL token or default to the `default` namespace.
+- `source_includes_var` - (bool: false) Whether or not the module configured at [`task.source`](#source) includes the [`consul_kv` variable](/docs/nia/terraform-modules#consul-kv-variable). Please refer to the documentation of the selected module for guidance on how to configure this field. If configured inconsistently with the module, Consul-Terraform-Sync will error and exit.
```

Comment:
Just want to make sure I'm understanding this one correctly:

So `source_includes_var` is only true if the module includes `consul_kv variable` and is always false otherwise?

Reply:
Yup! This is the flip side of what I mentioned in the other comment, if you pass a variable that the module does not expect, Terraform will also throw an error.

Reply:
```suggestion
- `source_includes_var` - (bool: false) If set to `true`, the module specified in the [`task.source`](#source) field will include the [`consul_kv` variable](/docs/nia/terraform-modules#consul-kv-variable). This field will default to `false` if the `consul_kv variable` is not configured or configured inconsistently. Refer to the documentation for the selected module for guidance on how to configure this field. If configured inconsistently with the module, Consul-Terraform-Sync will error and exit.
```
Just trying to integrate what @wilkermichael was asking about.

Reply:
Made a modification similar to your suggestion!